include/NeoFOAM/mesh/unstructuredMesh/unstructuredMesh.hpp

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: 2023 NeoFOAM authors
+#pragma once
+
+#include <array>
+#include <vector>
+
+#include "NeoFOAM/primitives/vector.hpp"
+#include "NeoFOAM/fields/FieldTypeDefs.hpp"
+#include "NeoFOAM/mesh/unstructuredMesh/BoundaryMesh.hpp"
+#include "NeoFOAM/mesh/stencil/StencilDataBase.hpp"
+
+namespace NeoFOAM
+{
+
+class unstructuredMesh
+{
+public:
+
+    unstructuredMesh(
+        vectorField points,
+        scalarField cellVolumes,
+        vectorField cellCentres,
+        vectorField faceAreas,
+        vectorField faceCentres,
+        scalarField magFaceAreas,
+        labelField faceOwner,
+        labelField faceNeighbour,
+        int32_t nCells,
+        int32_t nInternalFaces,
+        int32_t nBoundaryFaces,
+        int32_t nBoundaries,
+        int32_t nFaces,
+        BoundaryMesh boundaryMesh
+    );
+
+    const vectorField& points() const;
+
+    const scalarField& cellVolumes() const;
+
+    const vectorField& cellCentres() const;
+
+    const vectorField& faceCentres() const;
+
+    const vectorField& faceAreas() const;
+
+    const scalarField& magFaceAreas() const;
+
+    const labelField& faceOwner() const;
+
+    const labelField& faceNeighbour() const;
+
+    int32_t nCells() const;
```

Comment:
Would it make sense to wrap the integer type, aka add a label type?

Reply:
Ok I've just noticed you have labels, so they should be used here.

```suggestion
    label nCells() const;
```